### PR TITLE
Ports Kharmaani Alates from Baycode

### PR DESCRIPTION
**Pull request by HungryMoth**
Carmoney
carmoney2

     

An exceedingly ambitious port job from someone with the collective coding experience of one single afternoon and a class in high-school, this is my attempt at porting a basic form of the Kharmaani from bay. Lore is obviously changed from Bay's more antag focused Ascent Faction. As well as some of the unbalanced OP stats/abilities loosened.

**WIP-Lore**
Simplistic and open to improvement, tried to leave room for character development or changes by any others, as well as peoples personal takes on characters.

"Relatively new to the Periphary, the Kharmaani are a race of reclusive crystalline mantids. Kharmaan Alates are the most common and currently only face of the Ascent in Virgo-Erigone. With a culture of uncompromising servitude beaten into them with bladed forelimbs and a biology prone to eusociality, many Alates have found the relative isolation from the domineering larger castes and the drive to follow orders to be freeing, and have proven themselves remarkably sociable and varied in personality and drive, if a bit eccentric."
	
Alates are the lowest rung of Ascent society, and Virgo has proven itself a bit of a getaway for outcasts/loners of their kind. Finding a niche in their specialized talents with group work due to eusocial society, as well as a release in such a more open and free environment. (To eat people whoaaaa and do other things so coool.)

**Adds:**
1.  Kharmaan Alates - Mantid-esque crystalline aliens. (Not Whitelisted, open to changing.)
2. Some clothes and loadout options (Clothes) that fit them, considering all other clothing options around wouldn't work.
3. Mantid Languages.

**What this does NOT include due to lack of skill:**
(But should **probably** be done if possible.)
1. Kharmaan Gynes - The Giant Mantid Leaders, for lack of a better term. Probably best saved for admin events, so I started small. Probably the next simplest step to port.
carmoney3

2. Some of the more interesting species mechanics that just aren't easily ported from one codebase to the next, such as razorweb-weaving, crystalline body mechanics that reflect lasers, snowflake chems, full on RIGsuits for them. Wasn't a focus due to them not intended to have that kind of gear anyways, they aren't an antag faction here.
carmoney4

3. Varied Customization / Colorable Sprites / More Clothing Options (They currently are only blue, and only have an undersuit and a space suit. In the future it would be nice to see lots of different colored Mantids. Not a lot of room for too much customization at the moment between individuals.)

carmoney clothes

4. Kharmaani do not breathe pesticide or a unique atmos here. I debated making them phoron breathers and porting their fancy gas tanks to go with it, but i'm not really sure if it's needed or desired. I'm fully up to adding it if it's wanted though. Might make em a bit more exotic feeling.

**Final Notes:**
1. These aren't the same antags from Bay, they are not intended to be an antag faction by any means, our lore is completely disconnected from Bays' lore.
2. Speaking of the Lore, it's a WIP and is pretty empty for the moment, I left it open for expansion and adjustments by anyone interested.
3. Balance is probably all over the place. I tried my best in cutting the more uneeded and OP mechanics. Feel free to request adjustments.
4. Whitelisted is not enabled but it's up to staff to decide. Most absurdly crazy features of them are gone, so whitelist might just be more to uphold an RP standard or prevent an overabundance of similiar characters especially since they will all look similiar. (Unless someone codes colorable Mantids)

tl;dr - I ramble a lot, I am not a coder. I just wanted to play as some cute bugs. I believe it benefits the server by essentially allowing a more alien and unique species for people to enjoy, and more options are always neat right?

I'm sure there are a lot of oversights on my part. Most of the more OP antag-prone abilities of them was not ported, but the stats may still be a bit strong in certain areas. Open to critique.

Quick Edit: I didn't restrict them from wearing any normal clothes, but most will probably look janky due to the non-humanoid sprite, but for the sake of customization left the option available.

### CODE REVIEW

**Comment by Wickedtemp** [comment]
Didn't the creator of the species kindly request that we not port the species?

**Comment by Novacat** [comment]
That was GAS. Kharmaani was made by Zuhayr who is pretty chill.

**Comment by Wickedtemp** [comment]
If they're okay with it then yeh seems neat.

**Comment by KasparoVy** [comment]
Zuhayr a legend

**Comment by Arokha** [comment]
> Didn't the creator of the species kindly request that we not port the species?

As Ascian said, that was GAS, and the creator had mentioned they didn't mind porting part of them, for the record. I can't remember exactly if it was the sprites, lore, or name they asked we not use.

**Comment by Novacat** [comment]
They asked us to not port the name, lore, and sprites, only the code was free to use, which we cannot do much with anyway since a lot of that code depends on brainmed.

File: code/modules/clothing/clothing_vr.dm [hunk 1]
```diff
@@ -124,7 +124,8 @@
 	sprite_sheets = list(
 		SPECIES_TESHARI = 'icons/mob/species/seromi/suit.dmi',
 		SPECIES_VOX = 'icons/mob/species/vox/suit.dmi',
-		SPECIES_WEREBEAST = 'icons/mob/species/werebeast/suit.dmi')
+		SPECIES_WEREBEAST = 'icons/mob/species/werebeast/suit.dmi',
+		SPECIES_MANTID_ALATE = 'icons/mob/species/mantid/onmob_suit_alate.dmi')
```
Comment (Novacat) [inline code/modules/clothing/clothing_vr.dm:128]: This should probably be renamed 'suit.dmi' for consistency sake

File: code/modules/clothing/clothing_vr.dm [hunk 2]
```diff
@@ -160,7 +161,8 @@
 	sprite_sheets = list(
 		SPECIES_TESHARI = 'icons/mob/species/seromi/uniform.dmi',
 		SPECIES_VOX = 'icons/mob/species/vox/uniform.dmi',
-		SPECIES_WEREBEAST = 'icons/mob/species/werebeast/uniform.dmi')
+		SPECIES_WEREBEAST = 'icons/mob/species/werebeast/uniform.dmi',
+		SPECIES_MANTID_ALATE = 'icons/mob/species/mantid/onmob_under_alate.dmi')
```
Comment (Novacat) [inline code/modules/clothing/clothing_vr.dm:165]: uniform.dmi here.

File: code/modules/mob/living/carbon/human/species/station/station_special_vr.dm [hunk 3]
```diff
@@ -485,3 +485,168 @@
...
+	toxins_mod =            0.4 // Not as biologically fragile as meatboys.
+	radiation_mod =         0.4 // Not as biologically fragile as meatboys.
+	flash_mod =               2 // Highly photosensitive.
+	brute_mod =             1.5 // Incredibly fragile.
+	burn_mod =              0.5 // Yet reflective.
+	darksight = 7
+
+	min_age =                18
+	max_age =                140 //
+	slowdown =               -0.2
```
Comment (Novacat) [inline code/modules/mob/living/carbon/human/species/station/station_special_vr.dm:541]: Might as well remove this since -0.2 is, due to our tickrate, not going to have any effect.

File: code/modules/mob/living/carbon/human/species/station/station_special_vr.dm [hunk 4]
```diff
@@ -485,3 +485,168 @@
...
+	rarity_value =            3
+	gluttonous =              2
```
Comment (Novacat) [inline code/modules/mob/living/carbon/human/species/station/station_special_vr.dm:543]: Remove gluttonous, since we do not use it in our codebase.

File: code/modules/mob/living/carbon/human/species/station/station_special_vr.dm [hunk 5]
```diff
@@ -485,3 +485,168 @@
...
+	siemens_coefficient =     0
```
Comment (Novacat) [inline code/modules/mob/living/carbon/human/species/station/station_special_vr.dm:544]: Duplicate var? Probably should be removed.

File: code/modules/mob/living/carbon/human/species/station/station_special_vr.dm (hunk 3 above)
Comment (Arokha) [reply code/modules/mob/living/carbon/human/species/station/station_special_vr.dm:541]: (Anything that isn't a multiple of the tickrate is no different from the tickrate and ours is 0.5)

File: code/modules/client/preference_setup/loadout/loadout_xeno_vr.dm [hunk 6]
```diff
@@ -88,4 +88,42 @@
...
+	display_name = "Ascent Voidsuit (Alate)"
+	path = /obj/item/clothing/suit/space/void/ascent
+	sort_category = "Xenowear"
+	whitelisted = SPECIES_MANTID_ALATE
+
+/datum/gear/head/mantid_space
+	display_name = "Ascent Helmet (Alate)"
+	path = /obj/item/clothing/head/helmet/space/void/ascent
+	sort_category = "Xenowear"
+	whitelisted = SPECIES_MANTID_ALATE
```
Comment (TheFurryFeline) [inline code/modules/client/preference_setup/loadout/loadout_xeno_vr.dm:129]: Due to #7510, these lines will need to include the new version to resolve a conflict issue. Easily done.. To resolve your conflict, click on the button and remove the >>>>master and >>>>HEAD since you're adding both these and whatever's causing the conflict.

Comment (HungryMoth) [reply code/modules/client/preference_setup/loadout/loadout_xeno_vr.dm:125]: Fixed

File: code/modules/mob/living/carbon/human/species/station/station_special_vr.dm (hunk 5 above)
Comment (HungryMoth) [reply code/modules/mob/living/carbon/human/species/station/station_special_vr.dm]: Fixed

File: code/modules/mob/living/carbon/human/species/station/station_special_vr.dm (hunk 4 above)
Comment (HungryMoth) [reply code/modules/mob/living/carbon/human/species/station/station_special_vr.dm]: Fixed

File: code/modules/mob/living/carbon/human/species/station/station_special_vr.dm (hunk 3 above)
Comment (HungryMoth) [reply code/modules/mob/living/carbon/human/species/station/station_special_vr.dm]: It was originally -1 on Bay so I just lowered it a random amount, if intervals are just .5 i'll stick it in the middle for a middle ground I guess.

File: code/__defines/color.dm [hunk 7]
```diff
@@ -18,3 +18,4 @@
 
 #define COLOR_ASTEROID_ROCK    "#735555"
 #define COLOR_GOLD             "#ffcc33"
+#define COLOR_DARK             "#4c535b"
```
[the remark this answers is not in the document]
Comment (HungryMoth) [reply code/__defines/color.dm]: Fixed

File: code/modules/clothing/clothing_vr.dm (hunk 1 above)
Comment (HungryMoth) [reply code/modules/clothing/clothing_vr.dm]: fixed by removing onmob_ from everything

File: code/modules/clothing/clothing_vr.dm (hunk 2 above)
Comment (HungryMoth) [reply code/modules/clothing/clothing_vr.dm]: fixed by removing onmob_ from everything

**Comment by HungryMoth** [comment]
OKAY! Hello.

I think i'm done.

Fixed all the above concerns, made it so they can't wear clothes that don't make sense on them, added the organ sprites.

I guess this is ready for people to look over again?

**Comment by Novacat** [comment]
Unfortunately, after a discussion on the issue, as well as a conversation with the creator of the species, it was decided to not port these over. I apologize for this.
